ui/src/commonMain/kotlin/com/materiiapps/gloom/ui/screens/settings/AccountSettingsScreen.kt

```diff
@@ -0,0 +1,186 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.outlined.Edit
+import androidx.compose.material.pullrefresh.pullRefresh
+import androidx.compose.material.pullrefresh.rememberPullRefreshState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.IconToggleButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.TopAppBarScrollBehavior
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.koin.getScreenModel
+import cafe.adriel.voyager.navigator.LocalNavigator
+import cafe.adriel.voyager.navigator.currentOrThrow
+import com.materiiapps.gloom.Res
+import com.materiiapps.gloom.ui.components.LargeToolbar
+import com.materiiapps.gloom.ui.components.RefreshIndicator
+import com.materiiapps.gloom.ui.components.settings.SettingsButton
+import com.materiiapps.gloom.ui.screens.auth.LandingScreen
+import com.materiiapps.gloom.ui.screens.root.RootScreen
+import com.materiiapps.gloom.ui.utils.navigate
+import com.materiiapps.gloom.ui.viewmodels.settings.AccountSettingsViewModel
+import com.materiiapps.gloom.ui.widgets.accounts.AccountItem
+import com.materiiapps.gloom.ui.widgets.accounts.SignOutButton
+import com.materiiapps.gloom.ui.widgets.accounts.SignOutDialog
+import dev.icerock.moko.resources.compose.stringResource
+
+class AccountSettingsScreen : Screen {
+
+    @Composable
+    @OptIn(
+        ExperimentalMaterial3Api::class, ExperimentalMaterialApi::class,
+        ExperimentalFoundationApi::class
+    )
+    override fun Content() {
+        val viewModel: AccountSettingsViewModel = getScreenModel()
+        val scrollBehavior = TopAppBarDefaults.exitUntilCollapsedScrollBehavior()
+        val nav = LocalNavigator.currentOrThrow
+
+        val pullRefreshState = rememberPullRefreshState(
+            refreshing = viewModel.isLoading,
+            onRefresh = { viewModel.loadAccounts() }
+        )
+
+        val accounts by remember(viewModel.authManager.accounts) {
+            derivedStateOf {
+                viewModel.authManager.accounts.values
+                    .toList()
+                    .sortedByDescending { viewModel.authManager.currentAccount?.id == it.id }
```

Comment:
this is still unstable
use immutable lists / immutable wrappers instead (like upvote)